cmd/kots/cli/install.go

```diff
@@ -614,3 +644,66 @@ func getHttpProxyEnv(v *viper.Viper) map[string]string {
 	return env
 
 }
+
+func isAppReady(namespace string) string {
+
+	log := logger.NewLogger()
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+
+	clientset, err := k8sutil.GetClientset(kubernetesConfigFlags)
+	if err != nil {
+		return ""
+	}
+	podName, err := k8sutil.FindKotsadm(clientset, namespace)
+	if err != nil {
+		return ""
+	}
+
+	localPort, errChan, err := k8sutil.PortForward(kubernetesConfigFlags, 0, 3000, namespace, podName, false, stopCh, log)
+	if err != nil {
+		return ""
+	}
+
+	go func() {
+		select {
+		case <-errChan:
+		case <-stopCh:
+		}
+	}()
+
+	// copied from another function app-get.go
+	authSlug, _ := auth.GetOrCreateAuthSlug(kubernetesConfigFlags, namespace)
+
+	url := fmt.Sprintf("http://localhost:%d/api/v1/apps", localPort)
```

Comment:
only this part and further should remain in this function, everything else should be moved to `pollApplicationStatus` (outside the for loop).

Reply:
fixed.